lib/bearerstrategy.js

```diff
@@ -308,7 +308,7 @@ Strategy.prototype.authenticate = function authenticateStrategy(req) {
             .concat(`&${aadutils.getLibraryVersionParameterName()}=${aadutils.getLibraryVersion()}`);
 
         // use the policy name as the metadata cache key
-        cacheKey = 'policy: ' + policyName;
+        cacheKey = 'policy: ' + self._options.policyName;
```

Comment:
Yes. B2C works for BearerStrategy, since we only consume the token and the policy is given in the config. I tested the change with sample. For OIDCStrategy we need to figure out the policy.